libs/utils/android/wlauto.py

```diff
@@ -0,0 +1,141 @@
+
+class Wlauto(Workload):
+    """
+    Wrapper class for workload-automation
+
+    Workload-automation (WA) already provides many workloads for Android
+    targets.
+
+    In order to use one of the WA workloads (`wa list workloads` tells you what
+    is available) you need to implement a <workload__class_name> class in a
+    <workload_name>.py file stored under libs/utils/android/workloads.
+
+    <workload_class_name> inherits Wlauto and has to implement the methods from
+    Workload and (eventually) Wlauto.
+
+    This way, the workload is visible by the Workload class that will list it
+    as available and will eventually check that the related package is
+    installed on the Android target.
+
+    :param test_env: Test Environment object
+    :type test_env: TestEnv object
+    """
+
+    # Setup logger
+    logger = logging.getLogger('Wlauto')
+    logger.setLevel(logging.INFO)
+
+    def __init__(self, test_env):
+        if not find_executable('wa'):
+            raise RuntimeError(
+                'wa not installed. Please, install it from %s', WA_REPO
+            )
+
+        super(Wlauto, self).__init__(test_env)
+
+    @staticmethod
+    def _wload_is_available(workload_name):
+        """
+        Check if the specified workload is available in workload-automation.
+
+        :param workload_name: Name of the workload
+        :type workload_name: str
+        """
+        logging.debug('%14s - Checking for %s',
+                      'Wlauto',
+                      workload_name)
+
+        if sp.check_output(['wa', 'list', '-n', workload_name, 'workloads']):
+            return True
+
+        logging.error('%14s - %s not available in WA', 'Wlauto', workload_name)
+        return False
+
+    @staticmethod
+    def wa_run(agenda):
+        """
+        Run workload automation using the specified agenda.
+
+        :param agenda: Path to the YAML file to be passed to
+            workload-automation. You can pass either an absolute path or a
+            filename. In the latter case the file must be under
+            android/workloads.
+        :type agenda: str
+        """
+        # Check if the specified path is absolute
+        if agenda.startswith('/'):
+            path_to_agenda = agenda
+        else:
+            # Assume only file name
+            path_to_agenda = os.path.join(workloads_dir, agenda)
+
+        logging.debug('%14s - Using agenda %s',
+                      'Wlauto',
+                      path_to_agenda)
+
+        # Create temprary output directory for WA
+        tmp_dir = tempfile.mkdtemp()
+        logging.debug('%14s - wa output dir is %s', 'Wlauto', tmp_dir)
+
+        # Execute workload-automation
+        p = sp.Popen(WA_RUN_CMD.format(path_to_agenda, tmp_dir).split(),
+                     stdout=sp.PIPE,
+                     stderr=sp.PIPE)
+        out, err = p.communicate()
+        logging.debug(out)
+        logging.debug(err)
+        return tmp_dir
```

Comment:
Should we rename this to results_dir... at the end is where WA preduces it's output, isn't it?